mage_ai/services/k8s/config.py

```diff
@@ -23,27 +35,82 @@ class K8sExecutorConfig(BaseConfig):
     namespace: str = DEFAULT_NAMESPACE
     resource_limits: Dict = None
     resource_requests: Dict = None
-    service_account_name: str = DEFAULT_SERVICE_ACCOUNT_NAME
+    service_account_name: str = None
+    config_file: str = CONFIG_FILE
+    # those configs are loaded from the config file
+    metadata: Dict = None
+    container: Dict = None
+    pod: Dict = None
+    pod_config: V1PodSpec = None
+    meta: V1ObjectMeta = None
 
     @classmethod
     def load(self, config_path: str = None, config: Dict = None):
         executor_config = super().load(config_path=config_path, config=config)
-        # if executor_config.resource_limits is not None and \
-        #         type(executor_config.resource_limits) is dict:
-        #     try:
-        #         executor_config.resource_limits = K8sResourceConfig.load(
-        #             config=executor_config.resource_limits,
-        #         )
-        #     except Exception:
-        #         traceback.print_exc()
-        #         executor_config.resource_limits = None
-        # if executor_config.resource_requests is not None and \
-        #         type(executor_config.resource_requests) is dict:
-        #     try:
-        #         executor_config.resource_requests = K8sResourceConfig.load(
-        #             config=executor_config.resource_requests,
-        #         )
-        #     except Exception:
-        #         traceback.print_exc()
-        #         executor_config.resource_requests = None
+
+        service_account_name = DEFAULT_SERVICE_ACCOUNT_NAME
+        volumes = []
+        image_pull_secrets = {}
+        if executor_config.pod:
+            service_account_name = (
+                executor_config.pod.get('service_account_name') or DEFAULT_SERVICE_ACCOUNT_NAME
+            )
+
+            if executor_config.pod.get('volumes'):
+                volumes += [V1Volume(**e) for e in executor_config.pod['volumes']]
+
+            image_pull_secrets = None
+            if executor_config.pod.get('image_pull_secrets'):
+                image_pull_secrets = V1LocalObjectReference(
+                    name=executor_config.pod['image_pull_secrets'])
+
+        # to keep backward compatibility
+        if executor_config.service_account_name:
+            service_account_name = executor_config.service_account_name
+
+        executor_config.namespace = DEFAULT_NAMESPACE
```

Comment:
For backward compatibility, If `namespace` is set in executor_config, we should just use it.

Reply:
It will be never null so we check if it's `DEFAULT_NAMESPACE` we can override otherwise we keep it ?

Reply:
You can set the default value of `namespace` to `None`.